src/main/java/org/opensearch/security/dlic/rest/api/InternalUsersApiAction.java

```diff
@@ -115,6 +125,32 @@ protected Endpoint getEndpoint() {
         return Endpoint.INTERNALUSERS;
     }
 
+    @Override
+    protected void handleGet(final RestChannel channel, RestRequest request, Client client, final JsonNode content) throws IOException {
+        final String resourcename = request.param("name");
+
+        SecurityDynamicConfiguration<?> configuration = load(getConfigName(), true);
+        filter(configuration);
+
+        String filterBy = request.param("filterBy", "all");
+
+        if (filterBy.equalsIgnoreCase("internal") || filterBy.equalsIgnoreCase("service")) {
+            userService.filterAccountsByType(configuration, filterBy);
```

Comment:
If an unsupported filter type is found, an error should be returned indicating the allowed values otherwise you won't know if you used the term `users` and it didn't return any results.